Orange/widgets/data/owrank.py

```diff
@@ -698,66 +537,38 @@ def create_scores_table(self, labels):
 
     @classmethod
     def migrate_settings(cls, settings, version):
-        if not version:
-            # Before fc5caa1e1d716607f1f5c4e0b0be265c23280fa0
-            # headerState had length 2
-            headerState = settings.get("headerState", None)
-            if headerState is not None and \
-                    isinstance(headerState, tuple) and \
-                    len(headerState) < 3:
-                headerState = (list(headerState) + [None] * 3)[:3]
-                settings["headerState"] = headerState
-
-
-class ScoreValueItem(QStandardItem):
-    """A StandardItem subclass for python objects.
-    """
-    def __init__(self, *args):
-        super().__init__(*args)
-        self.setFlags(Qt.ItemIsSelectable | Qt.ItemIsEnabled)
-
-    def __lt__(self, other):
-        model = self.model()
-        if model is not None:
-            role = model.sortRole()
-        else:
-            role = Qt.DisplayRole
-        my = self.data(role)
-        other = other.data(role)
-        if my is None:
-            return True
-        return my < other
-
-
-class MySortProxyModel(QSortFilterProxyModel):
-
-    @staticmethod
-    def comparable(val):
-        return val is not None and \
-               (isinstance(val, str) or float('-inf') < val < float('inf'))
-
-    def lessThan(self, left, right):
-        role = self.sortRole()
-        left_data = left.data(role)
-        if not self.comparable(left_data):
-            left_data = float('-inf')
-        right_data = right.data(role)
-        if not self.comparable(right_data):
-            right_data = float('-inf')
-        try:
-            return left_data < right_data
-        except TypeError:
-            return str(left_data) < str(right_data)
+        # If older settings, restore sort header to default
+        # Saved selected_rows will likely be incorrect
+        if version is None or version < 2:
```

Comment:
Why check for None?
When settings dict does not have the __version__ key, [0 is passed as version](https://github.com/biolab/orange3/blob/master/Orange/widgets/settings.py#L461)

Reply:
Backwards compatibility. https://github.com/biolab/orange3/commit/642e446aeb709c4da448d82ed62b68820c5a19ff 😎